apps/web/lib/hooks/useSlots.ts

```diff
@@ -118,51 +118,65 @@ export const useSlots = (props: UseSlotsProps) => {
       inviteeDate: date,
       workingHours: responseBody.workingHours,
       minimumBookingNotice,
+      eventLength,
     });
+    // const busyTimes = [
+    //   { start: new Date(2022, 2, 9, 12, 50, 0, 0), end: new Date(2022, 2, 9, 13, 50, 0, 0) },
+    // ];
     // Check for conflicts
+    const finalizationTime = times[times.length - 1].add(eventLength, "minutes");
     for (let i = times.length - 1; i >= 0; i -= 1) {
-      responseBody.busy.every((busyTime): boolean => {
-        const startTime = dayjs(busyTime.start);
-        const endTime = dayjs(busyTime.end);
-        // Check if start times are the same
-        if (times[i].isBetween(startTime, endTime, null, "[)")) {
-          times.splice(i, 1);
-        }
-        // Check if slot end time is between start and end time
-        else if (times[i].add(eventLength, "minutes").isBetween(startTime, endTime)) {
-          times.splice(i, 1);
-        }
-        // Check if startTime is between slot
-        else if (startTime.isBetween(times[i], times[i].add(eventLength, "minutes"))) {
-          times.splice(i, 1);
-        }
-        // Check if time is between afterBufferTime and beforeBufferTime
-        else if (
-          times[i].isBetween(
-            startTime.subtract(beforeBufferTime, "minutes"),
-            endTime.add(afterBufferTime, "minutes")
-          )
-        ) {
-          times.splice(i, 1);
-        }
-        // considering preceding event's after buffer time
-        else if (
-          i > 0 &&
-          times[i - 1]
-            .add(eventLength + afterBufferTime, "minutes")
-            .isBetween(
-              startTime.subtract(beforeBufferTime, "minutes"),
-              endTime.add(afterBufferTime, "minutes"),
-              null,
-              "[)"
-            )
-        ) {
-          times.splice(i, 1);
-        } else {
-          return true;
-        }
-        return false;
-      });
+      // Check if the final slot surpasses the availability end time
+      if (
+        times[i]
+          .add(eventLength + beforeBufferTime + afterBufferTime, "minutes")
+          .isAfter(finalizationTime, "minute")
```

Comment:
I would appreciate to see an explicit if/else if sentence, to me this looks like a if else hell.
`times[i]
          .add(eventLength + beforeBufferTime + afterBufferTime, "minutes")`
          as
 `currentTime.add(totalEventDurationMinutes, "minutes")`
 Or probably just short form 
 `if(currentEventWithTotalDuration.isAfter(finalizationTimeMinutes)`
I think we can increase declaration of constants but make our code more readable in this delicate operation

Reply:
Yeah, that's a good point. I think we can make this a tad more readable for other devs to make sense of it quicker.